packages/opentelemetry-types/src/trace/instrumentation/Plugin.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Tracer } from '../tracer';
+
+/** Interface Plugin to apply patch. */
+export interface Plugin {
+  /**
+   * Method that enables the instrumentation patch.
+   * @param moduleExports nodejs module exports from the module to patch
```

Comment:
Could you give an example of this?

Reply:
High level stuff, something like this:
```
const DEFAULT_INSTRUMENTATION_MODULES = ['http', 'https', ...];
export class AutomaticTracer implements types.Tracer {
  constructor(){
    this.loadPlugins(DEFAULT_INSTRUMENTATION_MODULES);
  }

  loadPlugins(modulesToPatch: string[]) {
    // load plugin using its instrumentation plugin package name ie. opentelemetry-instrumentation-http.
    // Each plugin module should implement the Plugin interface and export an instance named as "plugin".
    const plugin: Plugin = require(moduleName as string).plugin;

    // Hook into the modules
    Hook(modulesToPatch, function (exports, name, basedir) {
      return plugin.enable(exports, tracer);
    }
  }
}
```

_* Name can be either AutomaticTracer or just Tracer or something else._

Reply:
Thanks - I actually just meant, what would be an example value for "moduleExports"? And would it make sense to add that to the JSDoc comment for the param?

Reply:
I don't think it can be typed since it's what the patched module export as object

Reply:
Yes, makes sense on not giving it a strict type - I mostly just wanted to understand how the parameter is used and the examples clarify that!